hw/src/trezor.rs

```diff
@@ -407,6 +404,41 @@ impl Manager {
 	}
 }
 
+pub struct EventHandler {
+	trezor: Weak<Manager>,
+}
+
+impl EventHandler {
+	pub fn new(trezor: Weak<Manager>) -> Self {
+		Self { trezor: trezor }
+	}
+}
+
+impl libusb::Hotplug for EventHandler {
+	fn device_arrived(&mut self, _device: libusb::Device) {
+		println!("trezor arrived");
+		if let Some(t) = self.trezor.upgrade() {
+			// Wait for the device to boot-up
+			thread::sleep(Duration::from_millis(1000));
```

Comment:
this seems arbitrary and prone to race conditions. is there a better way?

Reply:
Hrm, only a physical device TREZOR device can generate this and the ```update_devices()``` iterates through the devices and read/write to them after acquiring a RwLock. Thus, the risk is pretty low that somebody plugs in two TREZOR keys at the same time

But still ```Weak::upgrade() -> Rc``` is not atomic, so it that what you refer to? That could probably be solved by another lock!

Thoughts/suggestions?